### ISSUE TITLE
Hasura Migration - FatalError: syntax error at or near "AS"

### ISSUE THREAD

**Issue by ChakshuGautam** [hasura/graphql-engine#7401]
I am trying to migrate my local schema to our cluster in prod and it is throwing this error - `[42601] FatalError: syntax error at or near "AS"`

I am sharing the up.sql file as well here. 
```sql
CREATE FUNCTION public.set_current_timestamp_updated_at() RETURNS trigger
    LANGUAGE plpgsql
    AS $$
DECLARE
  _new record;
BEGIN
  _new := NEW;
  _new."updated_at" = NOW();
  RETURN _new;
END;
$$;
CREATE TABLE public.adapter (
    id uuid DEFAULT public.gen_random_uuid() NOT NULL,
    channel text NOT NULL,
    provider text NOT NULL,
    config jsonb NOT NULL,
    name text NOT NULL,
    updated_at timestamp with time zone DEFAULT now(),
    created_at timestamp with time zone DEFAULT now()
);
CREATE TABLE public.board (
    id integer NOT NULL,
    name text NOT NULL
);
CREATE SEQUENCE public.boards_id_seq
    AS integer
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;
ALTER SEQUENCE public.boards_id_seq OWNED BY public.board.id;
CREATE TABLE public.bot (
    id uuid DEFAULT public.gen_random_uuid() NOT NULL,
    name text NOT NULL,
    "startingMessage" text NOT NULL,
    users text[] NOT NULL,
    "logicIDs" text[] NOT NULL,
    owners text[],
    created_at timestamp with time zone DEFAULT now(),
    updated_at timestamp with time zone DEFAULT now(),
    status text,
    description text,
    "startDate" date,
    "endDate" date,
    purpose text
);
CREATE TABLE public."conversationLogic" (
    id uuid DEFAULT public.gen_random_uuid() NOT NULL,
    transformers jsonb NOT NULL,
    adapter uuid NOT NULL,
    name text NOT NULL,
    created_at timestamp with time zone DEFAULT now(),
    updated_at timestamp with time zone DEFAULT now(),
    description text
);
CREATE TABLE public.organisation (
    state text,
    district text,
    block text,
    school text,
    cluster text,
    id uuid DEFAULT public.gen_random_uuid() NOT NULL,
    created_at timestamp with time zone DEFAULT now(),
    updated_at timestamp with time zone DEFAULT now(),
    school_code text
);
CREATE TABLE public.pgbench_accounts (
    aid integer NOT NULL,
    bid integer,
    abalance integer,
    filler character(84)
)
WITH (fillfactor='100');
CREATE TABLE public.pgbench_branches (
    bid integer NOT NULL,
    bbalance integer,
    filler character(88)
)
WITH (fillfactor='100');
CREATE TABLE public.pgbench_history (
    tid integer,
    bid integer,
    aid integer,
    delta integer,
    mtime timestamp without time zone,
    filler character(22)
);
CREATE TABLE public.pgbench_tellers (
    tid integer NOT NULL,
    bid integer,
    tbalance integer,
    filler character(84)
)
WITH (fillfactor='100');
CREATE TABLE public.role (
    id integer NOT NULL,
    name text NOT NULL
);
CREATE SEQUENCE public.roles_id_seq
    AS integer
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;
ALTER SEQUENCE public.roles_id_seq OWNED BY public.role.id;
CREATE TABLE public.service (
    id uuid DEFAULT public.gen_random_uuid() NOT NULL,
    type text NOT NULL,
    config jsonb,
    created_at timestamp with time zone DEFAULT now(),
    updated_at timestamp with time zone DEFAULT now(),
    name text
);
CREATE TABLE public.transformer (
    name name NOT NULL,
    tags text[],
    config jsonb NOT NULL,
    id uuid DEFAULT public.gen_random_uuid() NOT NULL,
    service_id uuid NOT NULL,
    created_at timestamp with time zone DEFAULT now(),
    updated_at timestamp with time zone DEFAULT now()
);
CREATE TABLE public."userSegment" (
    id uuid DEFAULT public.gen_random_uuid() NOT NULL,
    name text NOT NULL,
    "all" uuid NOT NULL,
    "byID" uuid NOT NULL,
    "byPhone" uuid NOT NULL,
    created_at timestamp with time zone DEFAULT now(),
    updated_at timestamp with time zone DEFAULT now(),
    category text,
    count integer DEFAULT 0
);
ALTER TABLE ONLY public.board ALTER COLUMN id SET DEFAULT nextval('public.boards_id_seq'::regclass);
ALTER TABLE ONLY public.role ALTER COLUMN id SET DEFAULT nextval('public.roles_id_seq'::regclass);
ALTER TABLE ONLY public.adapter
    ADD CONSTRAINT adapter_name_key UNIQUE (name);
ALTER TABLE ONLY public.adapter
    ADD CONSTRAINT adapter_pkey PRIMARY KEY (id);
ALTER TABLE ONLY public.board
    ADD CONSTRAINT boards_pkey PRIMARY KEY (id);
ALTER TABLE ONLY public.bot
    ADD CONSTRAINT bot_name_key UNIQUE (name);
ALTER TABLE ONLY public.bot
    ADD CONSTRAINT bot_pkey PRIMARY KEY (id);
ALTER TABLE ONLY public.bot
    ADD CONSTRAINT "bot_startingMessage_key" UNIQUE ("startingMessage");
ALTER TABLE ONLY public."conversationLogic"
    ADD CONSTRAINT "conversationLogic_name_key" UNIQUE (name);
ALTER TABLE ONLY public."conversationLogic"
    ADD CONSTRAINT "conversationLogic_pkey" PRIMARY KEY (id);
ALTER TABLE ONLY public.organisation
    ADD CONSTRAINT organisation_pkey PRIMARY KEY (id);
ALTER TABLE ONLY public.pgbench_accounts
    ADD CONSTRAINT pgbench_accounts_pkey PRIMARY KEY (aid);
ALTER TABLE ONLY public.pgbench_branches
    ADD CONSTRAINT pgbench_branches_pkey PRIMARY KEY (bid);
ALTER TABLE ONLY public.pgbench_tellers
    ADD CONSTRAINT pgbench_tellers_pkey PRIMARY KEY (tid);
ALTER TABLE ONLY public.role
    ADD CONSTRAINT roles_pkey PRIMARY KEY (id);
ALTER TABLE ONLY public.service
    ADD CONSTRAINT "serviceType_pkey" PRIMARY KEY (id);
ALTER TABLE ONLY public.transformer
    ADD CONSTRAINT transformer_pkey PRIMARY KEY (id);
ALTER TABLE ONLY public."userSegment"
    ADD CONSTRAINT "userSegment_name_key" UNIQUE (name);
ALTER TABLE ONLY public."userSegment"
    ADD CONSTRAINT "userSegment_pkey" PRIMARY KEY (id);
CREATE INDEX bot_name ON public.bot USING btree (name);
CREATE INDEX "bot_startingMessage" ON public.bot USING btree ("startingMessage");
CREATE INDEX state_district ON public.organisation USING btree (state, district);
CREATE INDEX state_district_block ON public.organisation USING btree (state, district, block);
CREATE INDEX state_district_block_cluster ON public.organisation USING btree (state, district, block, cluster);
CREATE INDEX state_index ON public.organisation USING btree (state);
CREATE TRIGGER set_public_adapter_updated_at BEFORE UPDATE ON public.adapter FOR EACH ROW EXECUTE FUNCTION public.set_current_timestamp_updated_at();
COMMENT ON TRIGGER set_public_adapter_updated_at ON public.adapter IS 'trigger to set value of column "updated_at" to current timestamp on row update';
CREATE TRIGGER set_public_bot_updated_at BEFORE UPDATE ON public.bot FOR EACH ROW EXECUTE FUNCTION public.set_current_timestamp_updated_at();
COMMENT ON TRIGGER set_public_bot_updated_at ON public.bot IS 'trigger to set value of column "updated_at" to current timestamp on row update';
CREATE TRIGGER "set_public_conversationLogic_updated_at" BEFORE UPDATE ON public."conversationLogic" FOR EACH ROW EXECUTE FUNCTION public.set_current_timestamp_updated_at();
COMMENT ON TRIGGER "set_public_conversationLogic_updated_at" ON public."conversationLogic" IS 'trigger to set value of column "updated_at" to current timestamp on row update';
CREATE TRIGGER set_public_organisation_updated_at BEFORE UPDATE ON public.organisation FOR EACH ROW EXECUTE FUNCTION public.set_current_timestamp_updated_at();
COMMENT ON TRIGGER set_public_organisation_updated_at ON public.organisation IS 'trigger to set value of column "updated_at" to current timestamp on row update';
CREATE TRIGGER set_public_service_updated_at BEFORE UPDATE ON public.service FOR EACH ROW EXECUTE FUNCTION public.set_current_timestamp_updated_at();
COMMENT ON TRIGGER set_public_service_updated_at ON public.service IS 'trigger to set value of column "updated_at" to current timestamp on row update';
CREATE TRIGGER set_public_transformer_updated_at BEFORE UPDATE ON public.transformer FOR EACH ROW EXECUTE FUNCTION public.set_current_timestamp_updated_at();
COMMENT ON TRIGGER set_public_transformer_updated_at ON public.transformer IS 'trigger to set value of column "updated_at" to current timestamp on row update';
CREATE TRIGGER "set_public_userSegment_updated_at" BEFORE UPDATE ON public."userSegment" FOR EACH ROW EXECUTE FUNCTION public.set_current_timestamp_updated_at();
COMMENT ON TRIGGER "set_public_userSegment_updated_at" ON public."userSegment" IS 'trigger to set value of column "updated_at" to current timestamp on row update';
ALTER TABLE ONLY public."conversationLogic"
    ADD CONSTRAINT "conversationLogic_adapter_fkey" FOREIGN KEY (adapter) REFERENCES public.adapter(id) ON UPDATE RESTRICT ON DELETE RESTRICT;
ALTER TABLE ONLY public.transformer
    ADD CONSTRAINT transformer_type_fkey FOREIGN KEY (service_id) REFERENCES public.service(id) ON UPDATE RESTRICT ON DELETE RESTRICT;
ALTER TABLE ONLY public."userSegment"
    ADD CONSTRAINT "userSegment_all_fkey" FOREIGN KEY ("all") REFERENCES public.service(id) ON UPDATE RESTRICT ON DELETE RESTRICT;
ALTER TABLE ONLY public."userSegment"
    ADD CONSTRAINT "userSegment_byID_fkey" FOREIGN KEY ("byID") REFERENCES public.service(id) ON UPDATE RESTRICT ON DELETE RESTRICT;
ALTER TABLE ONLY public."userSegment"
    ADD CONSTRAINT "userSegment_byPhone_fkey" FOREIGN KEY ("byPhone") REFERENCES public.service(id) ON UPDATE RESTRICT ON DELETE RESTRICT;

```

Complete error
```
FATA[0001] apply failed: [postgres-error] query execution failed ($.args[0].args)
File: '1625729348951_init/up.sql'
```

I am sharing the Remote server logs as well
```log
{"type":"http-log","timestamp":"2021-08-17T09:11:50.852+0000","level":"error","detail":{"operation":{"user_vars":{"x-hasura-role":"admin"},"error":{"path":"$.args.table","error":"table \"hdb_catalog.hdb_version\" does not exist","code":"not-exists"},"request_id":"2359975f-4fa8-44a5-bfdc-a1add78f38e2","response_size":102,"query":{"args":{"where":null,"offset":null,"order_by":null,"source":"default","columns":["hasura_uuid","cli_state"],"limit":null,"table":{"schema":"hdb_catalog","name":"hdb_version"}},"version":1,"type":"select"}},"request_id":"2359975f-4fa8-44a5-bfdc-a1add78f38e2","http_info":{"status":400,"http_version":"HTTP/1.0","url":"/v1/query","ip":"11.2.6.44","method":"POST","content_encoding":null}}}
{"type":"http-log","timestamp":"2021-08-17T09:11:50.852+0000","level":"info","detail":{"operation":{"query_execution_time":6.278492e-3,"user_vars":{"x-hasura-role":"admin"},"request_id":"3ad537e9-7b2b-47b7-abb2-cfe5048aefeb","response_size":70,"request_read_time":5.301e-6},"request_id":"3ad537e9-7b2b-47b7-abb2-cfe5048aefeb","http_info":{"status":200,"http_version":"HTTP/1.0","url":"/v1/query","ip":"11.2.6.44","method":"POST","content_encoding":"gzip"}}}
{"type":"http-log","timestamp":"2021-08-17T09:11:50.852+0000","level":"info","detail":{"operation":{"query_execution_time":1.2379973e-2,"user_vars":{"x-hasura-role":"admin"},"request_id":"76274e48-e4b6-45be-b7fa-5b72fb33477d","response_size":68,"request_read_time":5.3e-6},"request_id":"76274e48-e4b6-45be-b7fa-5b72fb33477d","http_info":{"status":200,"http_version":"HTTP/1.0","url":"/v1/query","ip":"11.2.6.44","method":"POST","content_encoding":"gzip"}}}
{"type":"http-log","timestamp":"2021-08-17T09:11:50.852+0000","level":"info","detail":{"operation":{"query_execution_time":4.1303e-5,"user_vars":{"x-hasura-role":"admin"},"request_id":"01ddd313-0a00-44fb-a129-d16524a4960e","response_size":185,"request_read_time":4.1e-6},"request_id":"01ddd313-0a00-44fb-a129-d16524a4960e","http_info":{"status":200,"http_version":"HTTP/1.0","url":"/v1alpha1/config","ip":"11.2.6.44","method":"GET","content_encoding":"gzip"}}}
```

Localhost
Hasura Version: v2.0.0-alpha.1
PSQL: 13 (Docker)

Remote (prod)
Hasura Version: v2.0.0-alpha.1
PSQL: 9.6 (Azure)

Remote Deployment done on Kubernetes using [these templates](https://github.com/project-sunbird/sunbird-devops/tree/release-4.2.0/kubernetes/helm_charts/core/gql)

Any help on how to debug this is welcome. I know this is due to PSQL version but I should be able to do this even with different PSQL version right?

**Comment by ChakshuGautam** [comment]
Additional Info
`select * from hdb_catalog.hdb_version;` does provide the response on the Hasura SQL.

![Screenshot 2021-08-17 at 4 04 17 PM](https://user-images.githubusercontent.com/7413816/129711376-3d9a2b32-d053-42bb-8187-177b20af8a6f.png)

**Comment by ChakshuGautam** [comment]
@tirumaraiselvan

**Comment by ChakshuGautam** [comment]
@tirumaraiselvan @rikinsk 

I have figured out that Hasura does not generate 9.6 compatible schemas in the following cases

1. [AS related issues when generating schema](https://stackoverflow.com/questions/49208448/error-in-creating-sequences-when-restoring-the-postgresql-database) - I fixed those by removing `AS integer` above.
2. `EXECUTE FUNCTION` to be replaced with `EXECUTE PROCEDURE`.

I was able to manually do it by changing it piece by piece and `skip-execution` at the end.

Am I missing something basic here?
